src/rgw/rgw_op.cc

```diff
@@ -5426,7 +5426,7 @@ bool RGWCopyObj::parse_copy_location(const std::string_view& url_src,
     params_str = url_src.substr(pos + 1);
   }
 
-  if (name_str[0] == '/') // trim leading slash
+  if (!name_str.empty() && name_str[0] == '/') // trim leading slash
```

Comment:
thanks. https://en.cppreference.com/w/cpp/string/basic_string_view/remove_prefix does say that `name_str.remove_prefix(1)` is undefined behavior if `name_str` is empty, so this looks correct 👍 

but ultimately an empty `name_str` should cause us to return false below:
```c++
  pos = dec_src.find('/');
  if (pos == string::npos)
    return false;
```
so i'd suggest that we instead add an early return for `name_str.empty()` before this line:
```diff
+  if (name_str.empty()) {
+    return false;
+  }
   if (name_str[0] == '/') // trim leading slash
```
does that look right to you?